test/endecode_test.c

```diff
@@ -1451,7 +1451,10 @@ int setup_tests(void)
         ADD_TEST_SUITE_LEGACY(ECExplicitTri2G);
 # endif
 # ifndef OPENSSL_NO_SM2
-        ADD_TEST_SUITE(SM2);
+        if (!is_fips_3_0_0) {
```

Comment:
This isn't adequate.  is_fips_3_0_0 will be true when the FIPS provider isn't present.

Reply:
Could there be a comment where this variable is declared to explain what its potential values mean? And: is there a better name for this? Even just `fips_3_0_0_status` (and some `#define`s) is likely to at least cause people to tread carefully - a variable with just "is-foo" is likely to have its semantics assumed